### DISCUSSION TITLE
Moving files between buckets

### DISCUSSION THREAD

**Discussion by marbor3** [supabase/supabase#20117] [Questions] [answered]
Hi!

Quick question.

Is it somehow possible to move files between buckets?
I couldn't find it in docs, tried writing rpc to change bucket id myself, but I guess it also needs to move on storage side ...

Why do I need this: I have files uploaded by users, those can be private or public, with w simple switch at anytime one way or another.
My first idea - have just two buckets for the app - one public and one private - with folders inside for every user, and then just move folder to another bucket if as switch is requested.

Other idea: have bucket per user - then I saw there's a method to just change privacy of a bucket, so it would work, btu it will result in thousands of buckets - I'm not sure if it's the correct way to go ...

Maybe you can help me out here:
1. If I'm able to move assets between buckets?
2. if not - is it ok - so no major performance issues if instead I will be creating huge amount of buckets and toggling their privacy

Thank you!

**Comment by GaryAustin1** [comment, marked as answer]
There is no API call to move files between buckets.
I can't think of a performance impact by lots of buckets because all policies are done on the objects table and the bucket name is just a column there.
It could be a maintenance pain if you don't come up with generic policies that don't depend on the bucket name.

**Reply by marbor3** [reply]
Thanks! 

ok, if we wont think of anything else on other layers maybe we'll try bucket that includes some kind of user identification
